### PR TITLE
OSOE-792: Removing double escaping

### LINKED ISSUE #319: Unsupported Media Type error in add-azure-application-insights-release-annotation (OSOE-792)
As the title says, the add-azure-application-insights-release-annotation action fails with the following error:
```
ERROR: Unsupported Media Type({
  "Message": "The request contains an entity body but no Content-Type header. The inferred media type 'application/octet-stream' is not supported for this resource."
})
```

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-792)

### PR DESCRIPTION
**Pull request by wAsnk**
[OSOE-792](https://lombiq.atlassian.net/browse/OSOE-792)
Fixes #319

[OSOE-792]: https://lombiq.atlassian.net/browse/OSOE-792?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by wAsnk** [comment]
I'm pretty sure this isn't a valid error here:
https://github.com/Lombiq/GitHub-Actions/actions/runs/7798765332/job/21268124561?pr=322

**Comment by wAsnk** [comment]
> I'm pretty sure this isn't a valid error here: https://github.com/Lombiq/GitHub-Actions/actions/runs/7798765332/job/21268124561?pr=322

Nvm, they are.

File: .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1 [hunk 1]
```diff
@@ -17,7 +17,10 @@ $annotation = @{
 
 # Encoding parenthesis to prevent the request from failing.
 $ApplicationInsightsResourceId = $ApplicationInsightsResourceId.Replace('(', '%28').Replace(')', '%29')
-$body = (ConvertTo-Json $annotation -Compress) -replace '(\\+)"', '$1$1"' -replace "`"", "`"`""
+
+# Double escaping is not needed anymore, for more info see: https://github.com/Azure/azure-cli/issues/15529#issuecomment-1211884315
+$body = ConvertTo-Json $annotation -Compress
+
```
Comment (BenedekFarkas) [inline .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1:23]: OK, this is weird: Running the modified code on PS 7.4.1 (same as the runners) gives me the "Unsupported Media Type" error. If I revert the changes, then it works, so it's the opposite of the runners. Adding the content-type header (JSON) results in a bad request.

**Review by BenedekFarkas — CHANGES_REQUESTED** [review]

File: .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1 (hunk 1 above)
Comment (wAsnk) [reply .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1:23]: I'm not sure what to do with that, tbh.

Comment (wAsnk) [reply .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1:23]: The only thing I can image is if there was an error we run the double escaped version (old one).

Comment (BenedekFarkas) [reply .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1:23]: Updated the API call (Invoke-AzRestMethod instead of Az CLI), because that works consistently with this code.

**Review by BenedekFarkas — APPROVED** [review]

File: .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1 (hunk 1 above)
Comment (wAsnk) [reply .github/actions/add-azure-application-insights-release-annotation/Add-ReleaseAnnotation.ps1:23]: Oohhh yeah, I've tested that also with the double escaped characters...
